presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -176,6 +182,17 @@ protected InternalResourceGroup(
         this.staticResourceGroup = staticResourceGroup;
         this.additionalRuntimeInfo = requireNonNull(additionalRuntimeInfo, "additionalRuntimeInfo is null");
         this.shouldWaitForResourceManagerUpdate = requireNonNull(shouldWaitForResourceManagerUpdate, "shouldWaitForResourceManagerUpdate is null");
+        initQueryLevelLimits();
+    }
+
+    @GuardedBy("root")
+    private void initQueryLevelLimits()
+    {
+        if (parent.isPresent()) {
+            setMaxExecutionTime(parent.get().getMaxExecutionTime());
+            setTotalMemoryLimit(parent.get().getTotalMemoryLimit());
+            setMaxCpuTime(parent.get().getMaxCpuTime());
+        }
```

Comment:
I believe that this is an attempt to propagate the parent limits to children. This would be invoked only when the resource group is created, where as ideally - we want to do this any time a configuration change happens. Lets discuss this more tomorrow - but think of all the situations - 

1. parent limit changed
2. resource group limit changed
3. parent limit smaller than resource group limit